clang-tools-extra/clangd/SemanticSelection.cpp

```diff
@@ -238,7 +231,13 @@ getFoldingRanges(const std::string &Code, bool LineFoldingOnly) {
     AddFoldingRange(Start, End, FoldingRange::REGION_KIND);
   }
 
+  auto Preprocessed = DirectiveStructure.stripDirectives(OrigStream);
```

Comment:
Please preserve the comment `// FIXME: Provide ranges in the disabled-PP regions as well.`

This is referring to providing the existing kinds of ranges (bracket pairs and multi-line comments) in disabled-PP regions, which we're still not doing, so the FIXME remains relevant.